### PR TITLE
Hw 01

### PR DESCRIPTION
**Pull request by niksaz**

### CODE REVIEW

File: 1/noisifier.py [hunk 1]
```diff
@@ -0,0 +1,76 @@
...
+
+
+def augment_speech_files(speech_dir, output_dir, beep_dir, back_dir):
+    if not os.path.exists(output_dir):
+        os.makedirs(output_dir)
+
+    for speech_filename in os.listdir(speech_dir):
+        speech_file = os.path.join(speech_dir, speech_filename)
+        speech_base = os.path.splitext(speech_filename)[0]
+        output_file = os.path.join(output_dir, speech_base + '-aug.wav')
```
Comment (Rauf-Kurbanov) [inline 1/noisifier.py]: You can use f-string instead of concating `f'{speech_base}-aug.wav'`

File: 1/noisifier.py [hunk 2]
```diff
@@ -0,0 +1,76 @@
+# Author: Mikita Sazanovich
+
+import os
```
Comment (Rauf-Kurbanov) [inline 1/noisifier.py]: It is optional but consider trying sometime to use Pathlib as a modern alternative

File: 1/noisifier.py [hunk 3]
```diff
@@ -0,0 +1,76 @@
...
+    beep_data = load_audio_sec(beep_file, sr=sr)
+    back_data = load_audio_sec(back_file, sr=sr)
+
+    N = len(speech_data)
+    beep_data = repeat_or_trim_to(beep_data, N)
+    back_data = repeat_or_trim_to(back_data, N)
+
+    assert N == len(beep_data) and N == len(back_data)
+
+    output = speech_data + 0.25 * beep_data + 0.25 * back_data
```
Comment (Rauf-Kurbanov) [inline 1/noisifier.py]: Move this magic constants as parameters

File: 1/noisifier.py [hunk 4]
```diff
@@ -0,0 +1,76 @@
...
+    librosa.output.write_wav(output_file, output, sr=sr)
+
+
+def augment_speech_files(speech_dir, output_dir, beep_dir, back_dir):
+    if not os.path.exists(output_dir):
+        os.makedirs(output_dir)
```
Comment (Rauf-Kurbanov) [inline 1/noisifier.py]: What you really need here is `os.makedirs(output_dir, exit_ok=True)`

**Review by Rauf-Kurbanov — APPROVED** [review]
Regardless of minot comments - good job overall
Task is accepted - full credit

File: 1/noisifier.py (hunk 1 above)
Comment (niksaz) [reply 1/noisifier.py]: Thanks.

File: 1/noisifier.py (hunk 2 above)
Comment (niksaz) [reply 1/noisifier.py]: Did not know about it. But it is really convenient as I may say.

File: 1/noisifier.py (hunk 3 above)
Comment (niksaz) [reply 1/noisifier.py]: Good suggestion.

File: 1/noisifier.py (hunk 4 above)
Comment (niksaz) [reply 1/noisifier.py]: Right.

**Comment by niksaz** [comment]
Thanks for the comments, @Rauf-Kurbanov!

**Pull request merged**